pkg_resources/__init__.py

```diff
@@ -2098,8 +2098,13 @@ def _handle_ns(packageName, path_item):
     # capture warnings due to #1111
     with warnings.catch_warnings():
         warnings.simplefilter("ignore")
-        loader = importer.find_module(packageName)
-
+        try:
+            loader = importer.find_module(packageName)
+        except AttributeError:
+            try:
+                loader = importer.find_spec(packageName).loader
+            except:
+                loader = None
```

Comment:
I don't like this `try`/`except`. Why is it done?

If it's because `find_spec` raises an error instead of returning `None` in the case that `importer.find_module` would normally return `None`, please narrow the exception catching to the specific exception raised in that instance. If it's just to avoid ever seeing an exception, drop this entirely - I'm not at all comfortable swallowing exceptions.

Reply:
I narrowed the exceptions to AttributeError catches in the latest commit. This dodges a `"find_spec" in dir(...)` check, which is an alternative if you'd prefer to avoid try/except entirely. Let me know if this addresses your concern.